File: Backend/Models/UserAppointmentBooking.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Models
{
    public class UserAppointmentBooking : Common
    {
        [Required(ErrorMessage = "Der skal være mindst en tilkoblet bruger")]
        public required int UserId { get; set; }
        public User? User { get; set; }

        [Required(ErrorMessage = "Der skal være mindst en tilkoblet aftale")]
        public required int AppointmentId { get; set; }
        public Appointment? Appointment { get; set; }
    }
}

File: flutter_app/lib/classes/objects/user_appointment_booking.dart
import 'appointment.dart';
import 'common.dart';
import 'user.dart';

class UserAppointmentBooking extends Common {
  UserAppointmentBooking({
    required this.id,
    required this.UserId,
    this.user,
    required this.AppointmentId,
    this.appointment,
    this.createdAt,
    this.updatedAt,
  });

  final String UserId;
  final User? user;

  final String AppointmentId;
  final Appointment? appointment;

  @override
  final String id;

  @override
  final DateTime? createdAt;

  @override
  final DateTime? updatedAt;

      factory UserAppointmentBooking.fromJson(Map<String, dynamic> json) {

    return UserAppointmentBooking(
      id: json['id'] as String,
      UserId: json['userId'] as String,
      user: json['user'] == null ? null :  User.fromJson(json['user']),
      AppointmentId: json['appointmentId'] as String,
      appointment: json['appointment'] == null ? null : Appointment.fromJson(json['appointment']),
      createdAt: DateTime.tryParse(json['createdAt']),
      updatedAt: DateTime.tryParse(json['updatedAt']),
    );
  }
}
